src/features/keystone/KeystoneAccountAssignScreen.tsx

```diff
@@ -0,0 +1,212 @@
+import AccountIcon from '@components/AccountIcon'
+import Box from '@components/Box'
+import CircleLoader from '@components/CircleLoader'
+import FabButton from '@components/FabButton'
+import SafeAreaBox from '@components/SafeAreaBox'
+import Text from '@components/Text'
+import TextInput from '@components/TextInput'
+import CheckBox from '@react-native-community/checkbox'
+import { useNavigation } from '@react-navigation/native'
+import { useColors, useSpacing } from '@theme/themeHooks'
+import React, { memo, useCallback, useMemo, useState } from 'react'
+import { useAsyncCallback } from 'react-async-hook'
+import { useTranslation } from 'react-i18next'
+import { KeyboardAvoidingView, Platform, StyleSheet } from 'react-native'
+import { useSafeAreaInsets } from 'react-native-safe-area-context'
+import base58 from 'bs58'
+import { CSAccountVersion } from '@storage/cloudStorage'
+import { hex } from '@coral-xyz/anchor/dist/cjs/utils/bytes'
+import { RootNavigationProp } from '../../navigation/rootTypes'
+import { useAccountStorage } from '../../storage/AccountStorageProvider'
+import { ImportAccountNavigationProp } from '../onboarding/import/importAccountNavTypes'
+import { CreateAccountNavigationProp } from '../onboarding/create/createAccountNavTypes'
+import { useKeystoneOnboarding } from './KeystoneOnboardingProvider'
+
+const KeystoneAccountAssignScreen = () => {
+  const onboardingNav = useNavigation<
+    ImportAccountNavigationProp & CreateAccountNavigationProp
+  >()
+  const rootNav = useNavigation<RootNavigationProp>()
+  const { t } = useTranslation()
+  const [alias, setAlias] = useState('')
+  const { keystoneOnboardingData } = useKeystoneOnboarding()
+  const insets = useSafeAreaInsets()
+  const spacing = useSpacing()
+  const colors = useColors()
+  const { hasAccounts, accounts } = useAccountStorage()
+  const [setAsDefault, toggleSetAsDefault] = useState(false)
+
+  const existingNames = useMemo(
+    () => accounts && new Set(Object.values(accounts).map((a) => a.alias)),
+    [accounts],
+  )
+  const accountStorage = useAccountStorage()
+  const { execute: handlePress, loading } = useAsyncCallback(async () => {
+    const getName = (index: number): string => {
+      const name = `${alias} ${index + 1}`
+      if (!existingNames?.has(name)) {
+        return name
+      }
+      return getName(index + 1)
+    }
+
+    const accountBulk = keystoneOnboardingData.accounts.map(
+      (account, index) => ({
+        alias: getName(index),
+        address: base58.encode(
```

Comment:
I don't think that this will work. `address` is expected to be a Helium L1 style address (gross, I know, we mostly use `solanaAddress` now). 

https://github.com/helium/helium-program-library/blob/master/packages/migration-service/src/index.ts#L75-L91

This code might be helpful as a reference.

Reply:
Because the Keystone cold wallet only exports the Solana address in its data, I assigned the address value to be the Solana public key.

Reply:
Helium's old L1 used ED25119 keys. That means for every solana pubkey, there's a valid helium l1 representation of that key. That's what the code I linked above does. You should not put the solana address here. You should convert that address to a helium address. Unfortunately this is just some tech debt in this code since it used to run against the helium l1

Reply:
resolved